src/sagemaker/estimator.py

```diff
@@ -547,6 +547,7 @@ def deploy(
                 )
             model = self._compiled_models[family]
         else:
+            kwargs["output_kms_key"] = self.output_kms_key
```

Comment:
I realize that I'm probably the source of this problem (with one of my recent code changes), but should we distinguish more between `output_kms_key` and `kms_key`? also, is there a case where someone might want to specify `output_kms_key` in `deploy`?

Reply:
When we repack the model the new model tar ball is uploaded to the same bucket. we need to use the same output_kms_key in the estimator, otherwise uploading the repacked tar ball will fail. 

I can't really think of any use case where you would want a different kms key since we don't allow user to specify a new bucket when deploy is called.

Reply:
maybe let's rename the `Model`'s `output_kms_key` to something like `model_data_kms_key`? (`model_kms_key`?)

Reply:
i will do model_kms_key